### PR TITLE
fix: await for worker task to complete and refactor nats and sqs mess…

### PR DESCRIPTION
**Pull request by debajyoti-truefoundry**
…age fetch code

### CODE REVIEW

File: async_processor/nats_pub_sub.py [hunk 1]
```diff
@@ -93,23 +93,22 @@ async def get_input_message(
         self,
     ) -> AsyncIterator[Optional[bytes]]:
         psub = await self._get_psub()
-        while True:
+        msgs = []
+        try:
```
Comment (nikp1172) [inline async_processor/nats_pub_sub.py:97]: will this context manager return more than one message??

Comment (nikp1172) [reply async_processor/nats_pub_sub.py:97]: ohh got it, we are looping in the worker itself

Comment (debajyoti-truefoundry) [reply async_processor/nats_pub_sub.py:97]: Yes, this was an over-complication. The worker loop will be the only loop for now.

File: async_processor/worker.py [hunk 2]
```diff
@@ -29,13 +28,11 @@ def __init__(self, worker_config: WorkerConfig, processor: Processor):
         self._output = worker_config.output_config.to_output()
         self._processor = processor
         self._healthy = True
-        signal.signal(signal.SIGTERM, self.stop)
```
Comment (nikp1172) [inline async_processor/worker.py:32]: why did we remove this

Comment (debajyoti-truefoundry) [reply async_processor/worker.py:32]: FastAPI lifespan will handle it.

Comment (debajyoti-truefoundry) [reply async_processor/worker.py:32]: https://github.com/truefoundry/async_service/pull/11/files#diff-97e48d2f2456d5b71489c9e2fb6292375ce696a38437e15108c24a54c351c3e1R94

Comment (debajyoti-truefoundry) [reply async_processor/worker.py:32]: https://fastapi.tiangolo.com/advanced/events/

**Review by nikp1172 — APPROVED** [review]

**Pull request merged**